src/Entity/IdeasWorkshop/Guideline.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace AppBundle\Entity\IdeasWorkshop;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use AppBundle\Entity\EntityEnabledTrait;
+use AppBundle\Entity\EntityNameTrait;
+use AppBundle\Entity\EntityPositionTrait;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+
+/**
+ * @ORM\Entity
+ *
+ * @UniqueEntity("name")
+ *
+ * @Algolia\Index(autoIndex=false)
+ */
+class Guideline
+{
+    use EntityNameTrait;
+    use EntityEnabledTrait;
+    use EntityPositionTrait;
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var ArrayCollection
+     *
+     * @ORM\OneToMany(targetEntity="Question", mappedBy="guideline")
+     */
+    private $questions;
+
+    public function __construct(string $name, bool $enable = false)
+    {
+        $this->setName($name);
+        $this->setEnable($enable);
+
+        $this->questions = new ArrayCollection();
+    }
+
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    public function addQuestion(Question $question): void
+    {
+        $this->questions->add($question);
```

Comment:
donc you need to test `if (!$this->questions->contains($question))` before adding it to the collection ?

Reply:
:heavy_check_mark: